reports-scheduler/src/main/java/com/amazon/opendistroforelasticsearch/reportsscheduler/job/ReportsSchedulerJobRunner.java

```diff
@@ -0,0 +1,221 @@
+        if(!(jobParameter instanceof JobParameter)) {
+            throw new IllegalStateException("Job parameter is not instance of JobParameter, type: "
+                    + jobParameter.getClass().getCanonicalName());
+        }
+
+        if(this.clusterService == null) {
+            throw new IllegalStateException("ClusterService is not initialized.");
+        }
+
+        if (this.threadPool == null) {
+            throw new IllegalStateException("ThreadPool is not initialized.");
+        }
+
+        Runnable runnable = () -> {
+
+            JobParameter parameter = (JobParameter) jobParameter;
+            String reportDefId = parameter.getReportDefinitionId();
+
+
+            Map<String, Object> jsonMap = new HashMap<>();
+            jsonMap.put("to_be_executed", true);
+
+
+
+            //TODO: Save to a new index with report_definition_id and a status of to_be_executed: true/false
+            // Once Kibana pulls this index and find this document, it will create/deliver report and update this
+            // document field "to_be_executed: false"
+
+            //TODO: may need to first search for document, if it exists, we only update the to_be_executed field to true
+            // If not, we may consider adding a is_locked: false field to initialize
+
+
+            UpdateRequest request = new UpdateRequest(JOB_QUEUE_INDEX_NAME, reportDefId)
+                    .doc(jsonMap);
+
+
+            client.update(request, new ActionListener<UpdateResponse>() {
+                @Override
+                public void onResponse(UpdateResponse updateResponse) {
+                    log.info("Existing job triggered and pushed to queue, waiting to be picked up by Kibana");
+                }
+
+                @Override
+                public void onFailure(Exception e) {
+                    if (e instanceof ElasticsearchException) {
+                        if (((ElasticsearchException) e).status() == RestStatus.NOT_FOUND) {
+                            jsonMap.put("report_definition_id", reportDefId);
+                            jsonMap.put("is_locked", false);
+                            IndexRequest indexRequest = new IndexRequest()
+                                    .index(JOB_QUEUE_INDEX_NAME)
+                                    .id(reportDefId)
+                                    .source(jsonMap);
+
+                            client.index(indexRequest, new ActionListener<IndexResponse>() {
+                                @Override
+                                public void onResponse(IndexResponse indexResponse) {
+                                    log.info("New job created and push to queue, waiting to be picked up by Kibana"); // TODO: better log message
+                                }
+
+                                @Override
+                                public void onFailure(Exception e) {
+                                    log.error(e.toString());
```

Comment:
If small description is added (different per condition) then when debugging, it would be easy to identify condition.

Reply:
Got your point. Added